sphinx/ext/autodoc/__init__.py

```diff
@@ -1107,9 +1108,10 @@ def annotate_to_first_argument(self, func: Callable, typ: Type) -> None:
         if len(sig.parameters) == 0:
             return
 
-        name = list(sig.parameters)[0]
-        if name not in func.__annotations__:
-            func.__annotations__[name] = typ
+        param_list = list(sig.parameters.values())
```

Comment:
How about `params` or `parameters` for the variable? I prefer not to use its structures for the varname.

Reply:
done